ENIGMAsystem/SHELL/Widget_Systems/Zenity/dialogs.cpp

```diff
@@ -108,23 +108,21 @@ static int show_message_helperfunc(string message) {
     dialog_caption = window_get_caption();
 
   string str_command;
-  string str_title;
-  string str_cancel;
+  string str_title = add_escaping(dialog_caption, true, " ");
+  string str_cancel = "--info --ok-label=OK ";
+  string str_icon = "\" --icon-name=dialog-information);";
   string str_echo = "echo 1";
 
-  if (message_cancel)
+  if (message_cancel) {
     str_echo = "if [ $? = 0 ] ;then echo 1;else echo -1;fi";
-
-  str_title = add_escaping(dialog_caption, true, " ");
-  str_cancel = "--info --ok-label=OK ";
-
-  if (message_cancel)
     str_cancel = "--question --ok-label=OK --cancel-label=Cancel ";
+    str_icon = "\" --icon-name=dialog-question);";
+  }
 
   str_command = string("ans=$(zenity ") +
   string("--attach=$(sleep .01;xprop -root 32x '\t$0' _NET_ACTIVE_WINDOW | cut -f 2) ") +
   str_cancel + string("--title=\"") + str_title + string("\" --no-wrap --text=\"") +
-  add_escaping(message, false, "") + string("\" --icon-name=dialog-information);") + str_echo;
+  add_escaping(message, false, "") + str_icon + str_echo;
```

Comment:
If these commands get much more complicated, please consider removing the quotes and brackets from `str_icon`, and any other variables you add like this. It's better to keep parentheses and semicolons in view of the command being assembled. Putting both quotes in the variable would make sense, but putting just one quote/parenthesis to match the same in a previously-appended string variable? That gets dangerous quickly.